### PR TITLE
Update ECRGate documentation

### LINKED ISSUE #9234: ECRGate has wrong definition and doesn't implement RZX(pi/2)
### Environment

- **Qiskit Terra version**: 0.22
- **Python version**:  all
- **Operating system**: all

### What is happening?

The `ECRGate` documentation says it is supposed to implement a `RZX(pi.2)` gate, which as described in [Sheldon et al. PRA 2016](https://journals.aps.org/pra/abstract/10.1103/PhysRevA.93.060302) as the pulse sequence
```
     ┌───────────┐┌───┐┌────────────┐┌───┐
q_0: ┤0          ├┤ X ├┤0           ├┤ X ├
     │  Rzx(π/4) │└───┘│  Rzx(-π/4) │└───┘
q_1: ┤1          ├─────┤1           ├─────
     └───────────┘     └────────────┘     
```
However the current definition of `ECRGate` is missing the last X gate in its decompostion and is implemented as 
```
     ┌───────────┐┌───┐┌────────────┐
q_0: ┤0          ├┤ X ├┤0           ├
     │  Rzx(π/4) │└───┘│  Rzx(-π/4) │
q_1: ┤1          ├─────┤1           ├
     └───────────┘     └────────────┘
```
which doesn't implement a `RZX(pi/2) = (II -1j * ZX)/sqrt(2)` gate, it instead implements the unitary `(IX - XY) /sqrt(2)` 

### How can we reproduce the issue?

```
qi.SparsePauliOp.from_operator(ECRGate())
SparsePauliOp(['IX', 'XY'],
              coeffs=[ 0.70710678+0.j, -0.70710678+0.j])
```

### What should happen?

If the ECR is supposed to be implemented as described in [Sheldon et al. PRA 2016](https://journals.aps.org/pra/abstract/10.1103/PhysRevA.93.060302) it should implement an RZXGate(pi/2).

### Any suggestions?

To make it match the paper definition, one just needs to add the final X gate added to its decomposition definition and update its matrix definition

### PR DESCRIPTION
**Pull request by ajavadia**
### Summary
Fix definition of ECRGate .. it doesn't implement RZX(pi/2) as pointed out in #9234 

Fixes #9234

### CODE REVIEW

File: qiskit/circuit/library/standard_gates/ecr.py [hunk 1]
```diff
@@ -21,7 +21,7 @@
 
 
 class ECRGate(Gate):
-    r"""An echoed RZX(pi/2) gate implemented using RZX(pi/4) and RZX(-pi/4).
+    r"""An echoed cross-resonance gate.
 
     This gate is maximally entangling and is equivalent to a CNOT up to
```
Comment (chriseclectic) [inline qiskit/circuit/library/standard_gates/ecr.py:26]: Can you also update the doc string to include that the gate implements the unitary `\frac{1}{\sqrt}{IX-XY}` so its more clear.

Comment (ajavadia) [reply qiskit/circuit/library/standard_gates/ecr.py:26]: done

File: qiskit/circuit/library/standard_gates/ecr.py [hunk 2]
```diff
@@ -21,11 +21,12 @@
...
     single-qubit pre-rotations. The echoing procedure mitigates some
     unwanted terms (terms other than ZX) to cancel in an experiment.
+    More specifically, this gate implements :math:`\frac{1}{\sqrt{2}}{IX-XY}`.
```
Comment (jakelishman) [inline qiskit/circuit/library/standard_gates/ecr.py:29]:
```suggestion
    More specifically, this gate implements :math:`\frac{1}{\sqrt{2}}(IX-XY)`.
```

Comment (chriseclectic) [reply qiskit/circuit/library/standard_gates/ecr.py:29]: With all the confusion of qiskit ordering we might want to explicitly write it in tensor product notation `I\otimes X` etc. I'm not sure what convention we use in our other gate docs, but as I wrote it I was using `I \otimes X`, and `X \otimes Y` correspond to `target \otimes control` or`Q1 \otimes Q0` as the matrix def returned by `to_matrix`

Comment (jakelishman) [reply qiskit/circuit/library/standard_gates/ecr.py:29]: I personally don't particularly mind either way there - I think using whatever string order `Pauli` uses would be fine to match - so I'm happy to defer to you. I was just correcting the delimiters in the LaTeX expression.

**Review by mtreinish — APPROVED** [review]

**Pull request merged**